src/index.ts

```diff
@@ -74,7 +72,12 @@ class FlakeCheckerAction {
   }
 
   async check(): Promise<number> {
-    const binaryPath = await this.idslib.fetchExecutable();
+    const sourceBinary = inputs.getStringOrNull("source-binary");
+
+    const binaryPath =
+      sourceBinary !== null && sourceBinary !== ""
+        ? sourceBinary
+        : await this.idslib.fetchExecutable();
```

Comment:
I wonder: should this be implemented in the fetchExecutable layer generally?

Reply:
The issue is that a `source-binary` parameter doesn't make sense in all of the Actions that use `detsys-ts`, and if you call `inputs.getStringOrNull("source-binary")` at that level, then the `source-binary` input has to be listed in `action.yml`, even in Actions where it doesn't make sense.